src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestDescriptor.cs

```diff
@@ -0,0 +1,142 @@
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SignedHttpRequestDescriptor"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// </remarks> 
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="signingCredentials">Defines the security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        /// <param name="signedHttpRequestCreationParameters">A set of parameters required for creating a SignedHttpRequest.</param>
+        /// <param name="callContext">An opaque context used to store work when working with authentication artifacts.</param> 
+        public SignedHttpRequestDescriptor(string accessToken, HttpRequestData httpRequestData, SigningCredentials signingCredentials, SignedHttpRequestCreationParameters signedHttpRequestCreationParameters, CallContext callContext) 
+        {
+            AccessToken = !string.IsNullOrEmpty(accessToken) ? accessToken : throw LogHelper.LogArgumentNullException(nameof(accessToken));
+            HttpRequestData = httpRequestData ?? throw LogHelper.LogArgumentNullException(nameof(httpRequestData));
+            SigningCredentials = signingCredentials ?? throw LogHelper.LogArgumentNullException(nameof(signingCredentials));
+            SignedHttpRequestCreationParameters = signedHttpRequestCreationParameters ?? throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestCreationParameters));
+            CallContext = callContext ?? throw LogHelper.LogArgumentNullException(nameof(callContext));
+        }
+
+        /// <summary>
+        /// Gets an access token that contains the 'cnf' claim.
+        /// </summary>
+        public string AccessToken { get; }
+
+        /// <summary>
+        /// An opaque context used to store work when working with authentication artifacts.
+        /// </summary>
+        public CallContext CallContext { get; }
+
+        ///<summary> Gets or sets a "cnf" claim value as a JSON string.</summary>
+        /// <remarks>
+        /// If <see cref="SignedHttpRequestCreationParameters.CreateCnf"/> flag is set to <c>true</c>, <see cref="CnfClaimValue"/> can be used 
+        /// as a "cnf" claim value when creating a SignedHttpRequest payload.
+        /// If <see cref="SignedHttpRequestCreationParameters.CreateCnf"/> flag is set to <c>true</c>, and <see cref="CnfClaimValue"/> is null or empty,
+        /// a "cnf" claim value will be derived from a <see cref="SigningCredentials"/>.<see cref="SecurityKey"/>.
+        /// </remarks>
+        public string CnfClaimValue { get; set; }
```

Comment:
Why do we allow this to be set?

Reply:
It's an extensibility point to support different type of keys.
It might be useful when EC keys are used on on targets other than netstandard2.0.

Another possible use case might be symmetric keys. A user could ask for a PoP protected access token using the symmetric key thumbprint as req_cnf claim. A symmetric key can't be included in plain text to a SHR payload, it has to be encrypted.
As we don't support this scenario out-of-the-box, users could manually create the outer (SHR's) cnf claim and set it as a jwe.